File: back-end/src/controllers/health.controller.js
const { testConnection } = require('../../config/database');
const asyncHandler = require('../utils/asyncHandler');
const responseHelper = require('../utils/responseHelper');

/**
 * Controller para rotas de health check
 */
const healthController = {
  /**
   * Verifica a saúde da API e conexão com o banco de dados
   */
  checkHealth: asyncHandler(async (req, res) => {
    const dbConnected = await testConnection();
    
    return responseHelper.success(res, {
      database: dbConnected ? 'connected' : 'disconnected',
      timestamp: new Date().toISOString(),
      uptime: process.uptime(),
    }, 'API está funcionando');
  }),
};

module.exports = healthController;

File: backend/api/controllers/health_controller.py
from django.db import connection
from rest_framework.decorators import api_view

from api.utils import response_helper


@api_view(["GET"])
def check_health(_request):
    try:
        with connection.cursor() as cursor:
            cursor.execute("SELECT NOW()")
            cursor.fetchone()
        db_status = "connected"
    except Exception:
        db_status = "disconnected"

    data = {
        "database": db_status,
        "timestamp": None,
        "uptime": None,
    }
    return response_helper.success(data, "API está funcionando")


